letsencrypt/renewer.py

```diff
@@ -166,6 +167,16 @@ def main(config=None, cli_args=sys.argv[1:]):
     # specify a config file on the command line, which, if provided, should
     # take precedence over this one.
     config.merge(configobj.ConfigObj(cli_config.renewer_config_file))
+    # Ensure that all of the needed folders have been created before continuing
+    uid = os.geteuid()
+    if (not os.path.isdir(cli_config.renewal_configs_dir) or
+        not os.path.isdir(cli_config.config_dir)):
+        print "Could not find config directory. Exiting. "
+    else:
+        le_util.make_or_verify_dir(
+                cli_config.work_dir, constants.CONFIG_DIRS_MODE, uid)
+        le_util.make_or_verify_dir(
+                cli_config.logs_dir, constants.CONFIG_DIRS_MODE, uid)
```

Comment:
`work_dir` and `logs_dir` should be created regardless, i.e. outside of any `if/else` block